src/bp-xprofile/bp-xprofile-filters.php

```diff
@@ -785,8 +785,8 @@ function bp_xprofile_validate_nickname_value( $retval, $field_id, $value, $user_
 	}
 
 	// only alpha numeric, underscore, dash
-	if ( ! preg_match( '/^([A-Za-z0-9-_\.]+)$/', $value ) ) {
-		return sprintf( __( 'Invalid %s. Only "a-z", "0-9", "-", "_" and "." are allowed.', 'buddyboss' ), $field_name );
+	if ( ! preg_match( '/^([A-Za-z0-9\'-_\.]+)$/', $value ) ) {
+		return sprintf( __( 'Invalid %s. Only "a-z", "0-9", "\'", "-", "_" and "." are allowed.', 'buddyboss' ), $field_name );
```

Comment:
@meetmakadiya In the error message no need to add `\` with `'`.

Reply:
@dipskakadiya sprintf starting with` ' `so we need to add` \` without` \` like `" ' "` through error .
http://i.imgur.com/9lE8Fo9.png
http://i.imgur.com/16Hmg8M.png

Reply:
@meetmakadiya Thanks for detailed info. Looks good 👍